third-culture-software/bhima
# Shipments updates

This PR does these updates to shipment handling
- Provides a way to mark a shipment as *delivered* to the destination depot (but not unpacked or stock entered).  Note that this  is optional but would be helpful to perform this immediately when a shipment arrives at the destination depot especially if it may take some time to unpack the shipment and enter it into the destination depot stock.
- Adds a [+New Shipment] button on the Shipments Registry for convenience
- Updates the Shipments documentation (docs/en/stock-management/movement.shipments.md
- Updates to the ready-to-ship and completed confirmation dialogs.  Added a new confirmation dialog for marking a shipment as delivered.

Closes https://github.com/IMA-WorldHealth/bhima/issues/6470

**TESTING**
- Use bhima_test
- Go to: Shipments > New Shipment
  - Create  simple shipment with one item (eg, code=MOT...)
  - Go through these steps with this shipment
    - Mark it ready to ship
    - Do Stock Exit on it (use the Shipments Registry action menu to simplify this)
    - Mark the shipment as Delivered
    - Do the Stock Entry on it (use the Shipments Registry action menu)
    - The shipment will automatically be marked complete
    - Check the Tracking Log (action menu) to verify the actions taken
- Create a new shipment (use the [+New Shipment] button in the Shipments Registry
   - Add an item with quantity > 1 (eg quinine)
   - Do Stock Exit on it (use the Shipments Registry action menu to simplify this)
   - SKIP marking the shipment as Delivered
   - Do a partial Stock Entry on it (use the Shipments Registry action menu) 
      - But edit the lot to reduce the quantity
   - The shipment should now show a status of "Partial"
   - Use the action menu to mark the shipment as completed anyway
   - Check the Tracking Log (action menu) to verify the actions taken


-----
[View rendered docs/en/stock-management/movement.shipments.md](https://github.com/jmcameron/bhima/blob/shipments-updates/docs/en/stock-management/movement.shipments.md)
